src/rgw/rgw_cr_rados.cc

```diff
@@ -268,12 +268,21 @@ int RGWRadosGetOmapKeysCR::send_request() {
   set_status() << "send request";
 
   librados::ObjectReadOperation op;
-  op.omap_get_vals2(marker, max_entries, entries, nullptr, &rval);
+  op.omap_get_vals2(marker, max_entries, entries, nullptr, nullptr);
 
   cn = stack->create_completion_notifier();
   return ref.ioctx.aio_operate(ref.oid, cn->completion(), &op, NULL);
 }
 
+int RGWRadosGetOmapKeysCR::request_complete()
+{
+  int r = cn->completion()->get_return_value();
+
+  set_status() << "request complete; ret=" << r;
```

Comment:
LGTM, I have searched `aio_operate` in RGW code to check whether there is similar problem. It seems that all async rados CR(RGWRadosSetOmapKeysCR, RGWRadosRemoveOmapKeysCR, RGWRadosRemoveCR, RGWRadosBILogTrimCR) already use `cn->completion()->get_return_value()`.  

maybe we should create a base class, avoid those duplicated code?

Reply:
> maybe we should create a base class, avoid those duplicated code?

that sounds reasonable, yeah. it would help distinguish the ops that use aio_operate vs async_rados